fortran/netcdf_get_nd_expanded.f90

```diff
@@ -0,0 +1,5880 @@
+
+  end interface
+
+  integer(c_int) :: cncid, cvarid, cstatus
+
+  integer(c_int8_t), dimension(:,:), pointer :: farrayp
+  type(c_ptr) :: carrayp
+
+  integer(c_size_t), target, dimension(7) :: localStart, localCount
+  integer(c_ptrdiff_t), target, dimension(7) :: localStride, localMap
+
+  type(c_ptr) :: startp, countp, stridep, mapp
+
+  integer :: numDims, numSize, numSizeP1, counter
+
+  integer(c_size_t) :: totalLocalCount
+  integer :: i, j
+  integer(c_size_t) :: t1
+  integer(c_ptrdiff_t) :: t2
+
+  cstatus = 0
+
+  cncid  = ncid
+  cvarid = varid - 1 ! subtract 1 to get c varid
+
+  ! Set local arguments to default values
+  numDims = size(shape(values))
+  if (present(start)) numDims = size(start)
+  localStart(:) = 0
+  localCount(:numDims) = shape(values)
+  localCount(numDims+1:) = 1
+  localStride(:) = 1
+
+  totalLocalCount = 1
+  do counter = 1, numDims
+    localMap(counter) = totalLocalCount
+    if (counter == numDims) exit
+    totalLocalCount = totalLocalCount * localCount(counter)
+  enddo
+
+  if (present(start))  localStart (:numDims) = start(:)-1
+  if (present(count))  localCount (:numDims) = count(:)
+  if (present(stride)) localStride(:numDims) = stride(:)
+  if (present(map))    localMap(:numDims) = map(:)
+
+  do i = 1, numDims/2
+    j = numDims - i + 1
+    t1 = localStart(i)
+    localStart(i) = localStart(j)
+    localStart(j) = t1
+  enddo
+  do i = 1, numDims/2
+    j = numDims - i + 1
+    t1 = localCount(i)
+    localCount(i) = localCount(j)
+    localCount(j) = t1
+  enddo
+  do i = 1, numDims/2
+    j = numDims - i + 1
+    t2 = localStride(i)
+    localStride(i) = localStride(j)
+    localStride(j) = t2
+  enddo
+  do i = 1, numDims/2
+    j = numDims - i + 1
+    t2 = localMap(i)
+    localMap(i) = localmap(j)
+    localMap(j) = t2
+  enddo
+
+  farrayp => values
+  carrayp = c_loc(farrayp) 
+
+  startp = c_loc(localStart)
+  countp = c_loc(localCount)
+
+  if (present(stride)) then
+    stridep = c_loc(localStride)
+    if (present(map)) then
+      mapp = c_loc(localMap)
+      cstatus = my_get_varm_schar(cncid, cvarid, startp, countp, stridep, mapp, carrayp)
+    else
+      cstatus = my_get_vars_schar(cncid, cvarid, startp, countp, stridep, carrayp)
+    endif
+  else
+    cstatus = my_get_vara_schar(cncid, cvarid, startp, countp, carrayp)
+  end if
+
+  call c_f_pointer(carrayp, farrayp, shape(values))
+
+  nf90_get_var_2D_OneByteInt = cstatus
+
+end function nf90_get_var_2D_OneByteInt
+
+function nf90_get_var_3D_OneByteInt(ncid, varid, values, start, count, stride, map)
+
+  use, intrinsic :: iso_c_binding
+
+  implicit none
+
+  integer :: nf90_get_var_3D_OneByteInt
+
+  integer, intent(in) :: ncid, varid
+  integer(c_int8_t), target, dimension(:,:,:), intent(out) :: values
+  integer, dimension(:), optional, intent(in) :: start, count, stride, map
+
+  interface
+
+    function my_get_vara_schar(ncid, varid, startp, countp, ip) bind(c, name='nc_get_vara_schar')
```

Comment:
I believe you could use a named interface, instead of repeating.

Reply:
Also, why is this needed? Aren't these bindings already available? Also, why not use the F77 versions, which already take care of all the Fortran/C interfaces?